src/OrchardCore.Modules/OrchardCore.PublishLater/Indexes/PublishLaterPartIndex.cs

```diff
@@ -0,0 +1,33 @@
+using System;
+using OrchardCore.ContentManagement;
+using OrchardCore.PublishLater.Models;
+using YesSql.Indexes;
+
+namespace OrchardCore.PublishLater.Indexes
+{
+    public class PublishLaterPartIndex : MapIndex
+    {
+        public DateTime? ScheduledPublishUtc { get; set; }
+    }
+
+    public class PublishLaterPartIndexProvider : IndexProvider<ContentItem>
+    {
+        public override void Describe(DescribeContext<ContentItem> context)
+        {
+            context.For<PublishLaterPartIndex>()
+                .Map(contentItem =>
+                {
+                    var publishLaterPart = contentItem.As<PublishLaterPart>();
+                    if (publishLaterPart != null)
```

Comment:
as @Skrypt says something like `|| !(contentItem.Published || contentItem.Latest) return null` (not sure of the exact logic you require) will stop this creating indexes for every item.

Reply:
Thanks, done that. Now there will be only an index record for the latest draft, and only if it actually has a date scheduled.